desktop/ui/src/main/java/org/datacleaner/user/MutableReferenceDataCatalog.java

```diff
@@ -213,13 +257,41 @@ public void removeStringPattern(StringPattern sp, boolean externalize) {
             for (StringPatternChangeListener listener : _stringPatternListeners) {
                 listener.onRemove(sp);
             }
+            for(ReferenceDataChangeListener<StringPattern> listener : _stringPatternV2Listeners){
+                listener.onRemove(sp);
+            }
         }
         if (externalize) {
             _configurationWriter.removeStringPattern(sp.getName());
             _userPreferences.save();
         }
     }
 
+    public void changeStringPattern(StringPattern oldPattern, StringPattern newPattern){
+        changeStringPattern(oldPattern, newPattern, true);
+    }
+    
+    public void changeStringPattern(StringPattern oldPattern, StringPattern newPattern, boolean externalize){
+        // The old reference is removed from user preferences and we add a new
+        // patern with the same name but with a
```

Comment:
patern should be pattern (and the comment line below can be appended to this one).